codalab/model/worker_model.py

```diff
@@ -0,0 +1,322 @@
+                                cl_worker.c.worker_id == worker_id))
+            ).fetchone()
+            if existing_row:
+                socket_id = existing_row.socket_id  
+                conn.execute(
+                    cl_worker.update()
+                        .where(and_(cl_worker.c.user_id == user_id,
+                                    cl_worker.c.worker_id == worker_id))
+                        .values(worker_row))
+                conn.execute(
+                    cl_worker_dependency.delete()
+                        .where(and_(cl_worker_dependency.c.user_id == user_id,
+                                    cl_worker_dependency.c.worker_id == worker_id)))   
+            else:
+                socket_id = self.allocate_socket(user_id, worker_id, conn)
+                worker_row.update({
+                    'user_id': user_id,
+                    'worker_id': worker_id,
+                    'socket_id': socket_id,
+                })
+                conn.execute(cl_worker.insert().values(worker_row))
+            dependency_rows = [{
+                'user_id': user_id,
+                'worker_id': worker_id,
+                'dependency_uuid': uuid,
+            } for uuid in dependency_uuids]
+            if dependency_rows:
+                conn.execute(cl_worker_dependency.insert(), dependency_rows)
+
+        return socket_id
+
+    def worker_cleanup(self, user_id, worker_id):
+        """
+        Deletes the worker and all associated data from the database as well
+        as the socket directory.
+        """
+        with self._engine.begin() as conn:
+            socket_rows = conn.execute(
+                cl_worker_socket.select()
+                    .where(and_(cl_worker_socket.c.user_id == user_id,
+                                cl_worker_socket.c.worker_id == worker_id))
+            ).fetchall()
+            for socket_row in socket_rows:
+                self._cleanup_socket(socket_row.socket_id)
+            conn.execute(cl_worker_socket.delete()
+                         .where(and_(cl_worker_socket.c.user_id == user_id,
+                                     cl_worker_socket.c.worker_id == worker_id)))   
+            conn.execute(cl_worker_run.delete()
+                         .where(and_(cl_worker_run.c.user_id == user_id,
+                                     cl_worker_run.c.worker_id == worker_id)))      
+            conn.execute(cl_worker_dependency.delete()
+                         .where(and_(cl_worker_dependency.c.user_id == user_id,
+                                     cl_worker_dependency.c.worker_id == worker_id)))      
+            conn.execute(cl_worker.delete()
+                         .where(and_(cl_worker.c.user_id == user_id,
+                                     cl_worker.c.worker_id == worker_id)))
+            
+    def get_workers(self):
+        """
+        Returns information about all the workers in the database.
```

Comment:
I would still write a sentence saying what the type of the return value is - it's a simple thing to do.